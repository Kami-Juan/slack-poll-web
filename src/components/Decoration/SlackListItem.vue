<template>
  <li>
    <input type="text" class="option-input-element" v-model="option" />
    <a class="option-remove-element" @click="remove">‚ùå</a>
  </li>
</template>
<script>
export default {
  props: ['value'],
  methods: {
    remove() {
      this.$emit('remove')
    }
  },
  computed: {
    option: {
      get() {
        return this.value
      },
      set(event) {
        this.$emit('input', event)
      }
    }
  }
}
</script>
<style scoped>
.option-input-element {
  border: none;
}

.option-input-element:focus {
  outline: 0;
}

.option-remove-element {
  cursor: pointer;
}
</style>