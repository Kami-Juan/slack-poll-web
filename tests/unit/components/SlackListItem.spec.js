import { mount } from '@vue/test-utils'
import SlackListItem from './../../../src/components/Decoration/SlackListItem'

describe('test SlackListItem.vue', () => {
  let cmp

  beforeEach(() => {
    cmp = mount(SlackListItem, {
      propsData: {
        value: 'One element'
      }
    })
  })

  it('should be render', () => {
    expect(cmp.element).toMatchSnapshot()
  })

  it('the option input has the correct value', () => {
    let optionInput = cmp.find('.option-input-element').element.value
    expect(optionInput).toEqual('One element')
  })

  it('the delete button has the correct emoji', () => {
    const deleteButton = cmp.find('.option-remove-element')
    expect(deleteButton.text()).toEqual('❌')
  })

  it('set data in the option input and fire the input event', () => {
    const optionInput = cmp.find('.option-input-element')
    optionInput.setValue('Second element')

    expect(cmp.emitted().input).toBeTruthy()
    expect(cmp.emitted().input.length).toBe(1)
    expect(cmp.emitted().input[0]).toEqual(['Second element'])
  })

  it('click the remove button and fire a event', () => {
    const deleteButton = cmp.find('.option-remove-element')
    deleteButton.trigger('click')

    expect(cmp.emitted().remove).toBeTruthy()
    expect(cmp.emitted().remove.length).toBe(1)
    expect(cmp.emitted().remove[0]).toEqual([])
  })
})