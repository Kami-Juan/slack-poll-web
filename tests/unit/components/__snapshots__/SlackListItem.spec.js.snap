// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test SlackListItem.vue should be render 1`] = `
<li>
  <input
    class="option-input-element"
    type="text"
  />
   
  <a
    class="option-remove-element"
  >
    ‚ùå
  </a>
</li>
`;
